cmd/sourced/dir/dir_test.go

```diff
@@ -0,0 +1,118 @@
+package dir
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidate(t *testing.T) {
+	assert := assert.New(t)
+
+	err := validate("/does/not/exists")
+	assert.True(ErrNotExist.Is(err))
+	assert.EqualError(err, "/does/not/exists does not exist")
+
+	// with a file
+	tmpFile := path.Join(os.TempDir(), "tmp-file")
+	f, err := os.Create(tmpFile)
+	assert.Nil(err)
+	assert.Nil(f.Close())
+	defer func() {
+		os.RemoveAll(tmpFile)
+	}()
+
+	err = validate(tmpFile)
+	assert.True(ErrNotValid.Is(err))
+	assert.EqualError(err, tmpFile+" is not a valid config directory: it is not a directory")
+
+	// with a dir
+	tmpDir := path.Join(os.TempDir(), "tmp-dir")
+	assert.Nil(os.Mkdir(tmpDir, os.ModePerm))
+	defer func() {
+		os.RemoveAll(tmpDir)
+	}()
+
+	err = validate(tmpDir)
+	assert.Nil(err)
+
+	// read only
+	assert.Nil(os.Chmod(tmpDir, 0444))
+	err = validate(tmpDir)
+	assert.True(ErrNotValid.Is(err))
+	assert.EqualError(err, tmpDir+" is not a valid config directory: it has no read-write access")
+
+	// write only
+	assert.Nil(os.Chmod(tmpDir, 0222))
+	err = validate(tmpDir)
+	assert.True(ErrNotValid.Is(err))
+	assert.EqualError(err, tmpDir+" is not a valid config directory: it has no read-write access")
+}
+
+func TestPrepare(t *testing.T) {
+	assert := assert.New(t)
+
+	tmpDir := path.Join(os.TempDir(), "tmp-dir")
+	assert.Nil(os.Mkdir(tmpDir, os.ModePerm))
+	defer func() {
+		os.RemoveAll(tmpDir)
+	}()
+
+	originSrcdDir := os.Getenv("SOURCED_DIR")
+	defer func() {
```

Comment:
Is this needed? Isn't it set only for the current process running the tests?

Reply:
Yes it is set only for the current process running the tests. But the value would change in the next test which can be unexpected. Example:

```go
// some_test.go
package main

import (
	"fmt"
	"os"
	"testing"
)

func TestOne(t *testing.T) {
	fmt.Println("test one env value: " + os.Getenv("SOME"))
	os.Setenv("SOME", "new value")
}

func TestTwo(t *testing.T) {
	fmt.Println("test two env value: " + os.Getenv("SOME"))
}
```

```
$ SOME=original-value go test some_test.go -v
=== RUN   TestOne
test one env value: original-value
--- PASS: TestOne (0.00s)
=== RUN   TestTwo
test two env value: new value
--- PASS: TestTwo (0.00s)
PASS
ok  	command-line-arguments	0.005s
```